spec/services/vacancy_facets_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+RSpec.describe VacancyFacets do
+  let(:store) { MockRedis.new }
+  let(:subject) { described_class.new(store: store) }
+
+  describe '#get' do
+    context 'when the field is set in redis' do
+      let(:field) { :job_roles }
+      let(:facet) { { teacher: 5 }.to_json }
+
+      before { store.set(field, facet) }
+
+      it 'retrieves the relevant facet' do
+        expect(subject.get(field)).to eq({ 'teacher' => 5 })
+      end
+    end
+
+    context 'when the field is not been set in redis' do
+      let(:field) { :not_set }
+
+      it 'returns an empty hash' do
+        expect(subject.get(field)).to eq({})
+      end
+    end
+  end
+
+  describe '#refresh' do
+    it 'calls Search::VacancySearchBuilder' do
+      JOB_ROLES_SUBJECTS_CITIES_AND_COUNTIES = 229
```

Comment:
This seems a bit flaky

Reply:
Changed this now. See what you think after iv'e pushed